cli/cmd/install-cni-plugin.go

```diff
@@ -1,101 +1,44 @@
 package cmd
 
 import (
+	"bytes"
 	"fmt"
 	"io"
 	"os"
-	"strings"
+	"path"
+	"time"
 
 	"github.com/linkerd/linkerd2/pkg/charts"
 	cnicharts "github.com/linkerd/linkerd2/pkg/charts/cni"
 	"github.com/linkerd/linkerd2/pkg/charts/static"
 	"github.com/linkerd/linkerd2/pkg/cmd"
 	"github.com/linkerd/linkerd2/pkg/flags"
 	"github.com/linkerd/linkerd2/pkg/version"
-	log "github.com/sirupsen/logrus"
+	valuespkg "helm.sh/helm/v3/pkg/cli/values"
 	"github.com/spf13/cobra"
 	"helm.sh/helm/v3/pkg/chart/loader"
 	"helm.sh/helm/v3/pkg/chartutil"
-	"sigs.k8s.io/yaml"
+	"helm.sh/helm/v3/pkg/engine"
 )
 
-const (
-	helmCNIDefaultChartName = "linkerd2-cni"
-	helmCNIDefaultChartDir  = "linkerd2-cni"
-)
-
-type cniPluginImage struct {
-	name       string
-	version    string
-	pullPolicy interface{}
-}
-
```

Comment:
Before we remove the old options, we should first try to get everything working together, similar to what we're doing for `install.go`. I'm not extremely comfortable with deleting a huge chunk of code without any deprecation warnings, even if we don't necessarily expect people to use the CLI as part of their production workflows.

Before we start to replace everything entirely, let's first see what it would look like to _add_ additional behaviour here. I suspect it won't complicate the code too much. That will also fix the tests :) 

We need to have a discussion to see if we want to introduce a breaking change here.

Reply:
I think I will have to rethink about implementation of `--set `flag. I was trying to follow the way it is done in `install.go` and how it leverages `values.go` file.